File: aio/curry.py
curryin = open("curryin.txt", "r")
curryout = open("curryout.txt", "w")
 
nums = list(map(int, curryin.readline().split()))
spoonfuls = [0, 0, 0]
 
while True:
    if nums[0] >= nums[1] and nums[2] >= nums[1]:
        spoonfuls[1] += 1
        nums[0], nums[2] = nums[0] - 1, nums[2] - 1
    
    elif nums[1] >= nums[0] and nums[2] >= nums[0]:
        spoonfuls[0] += 1
        nums[1], nums[2] = nums[1] - 1, nums[2] - 1
 
    else:
        spoonfuls[2] += 1
        nums[0], nums[1] = nums[0] - 1, nums[1] - 1
 
    if sum(nums) == nums[0] or sum(nums) == nums[1] or sum(nums) == nums[2]:
        break
 
string = str(spoonfuls[0])+" "+str(spoonfuls[1])+" "+str(spoonfuls[2])
curryout.write(string)
curryout.close()

File: aio/curry.cpp
#include <iostream>
#include <cstdio>

bool check(int a, int b, int c) {
    if (a >= 1 and b >= 1) {
        if (a >= c and b >= c) {
            return true;
        }
    }
    return false;
}

int main() {
    freopen("curryin.txt", "r", stdin);
    freopen("curryout.txt", "w", stdout);

    int a, b, c;
    int x = 0;
    int y = 0;
    int z = 0;

    std::cin >> a >> b >> c;

    while (check(a, b, c) or check(c, b, a) or check(a, c, b)) {
        if (check(a, b, c)) {
            z ++; a --; b --;
        }

        if (check(c, b, a)) {
            x ++; c--; b--;
        }

        if (check(a, c, b)) {
            y ++; a--; c--;
        }
    }

    std::cout << x << " " << y << " " << z;
    
    return 0;
}
